# Pandas Tips and Ticks
written by: Dr. Allah Ditta Babar\
date: 07/08/2026\
Email: adbabar.nibge@gmail.com

In [35]:
#load dataset to work with pandas
# Seaborn has data sets
import pandas as pd
import seaborn as sns
# titanic data
kashti= sns.load_dataset('titanic')
kashti.head()


,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [36]:
kashti.info() # rows are lso known as instences

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


In [37]:
# what every column means here
kashti.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

| Column            | Explanation                                                | Example                                  |
| ----------------- | ---------------------------------------------------------- | ---------------------------------------- |
| **`survived`**    | Whether the passenger survived                             | `1` = survived, `0` = died               |
| **`pclass`**      | Passenger's ticket class                                   | `1` = First, `2` = Second, `3` = Third   |
| **`sex`**         | Passenger's sex                                            | `male`, `female`                         |
| **`age`**         | Passenger's age in years                                   | `22.0`, `38.0`                           |
| **`sibsp`**       | Number of siblings or spouses traveling with the passenger | `0`, `1`, `2`                            |
| **`parch`**       | Number of parents or children traveling with the passenger | `0`, `1`, `2`                            |
| **`fare`**        | Price of the passenger's ticket                            | `7.25`, `71.28`                          |
| **`embarked`**    | Port where the passenger boarded                           | `C`, `Q`, `S`                            |
| **`class`**       | Passenger class as text                                    | `First`, `Second`, `Third`               |
| **`who`**         | Passenger category                                         | `man`, `woman`, `child`                  |
| **`adult_male`**  | Whether the passenger was an adult male                    | `True`, `False`                          |
| **`deck`**        | Deck where the passenger's cabin was located               | `A`, `B`, `C`, etc.                      |
| **`embark_town`** | Full name of boarding town                                 | `Cherbourg`, `Queenstown`, `Southampton` |
| **`alive`**       | Survival status as text                                    | `yes`, `no`                              |
| **`alone`**       | Whether the passenger was traveling alone                  | `True`, `False`                          |


In [38]:
# how many data sets are there in sns.load_datasets
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [45]:
len(sns.get_dataset_names())

22

In [42]:
for i, dataset in enumerate(sns.get_dataset_names(), 1):
    print(i, dataset)

1 anagrams
2 anscombe
3 attention
4 brain_networks
5 car_crashes
6 diamonds
7 dots
8 dowjones
9 exercise
10 flights
11 fmri
12 geyser
13 glue
14 healthexp
15 iris
16 mpg
17 penguins
18 planets
19 seaice
20 taxis
21 tips
22 titanic


In [48]:
kashti.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


In [47]:
kashti.tail()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
886,0,2,male,27.0,0,0,13.00,S,Second,man,True,NaN,Southampton,no,True
887,1,1,female,19.0,0,0,30.00,S,First,woman,False,B,Southampton,yes,True
888,0,3,female,NaN,1,2,23.45,S,Third,woman,False,NaN,Southampton,no,False
889,1,1,male,26.0,0,0,30.00,C,First,man,True,C,Cherbourg,yes,True
890,0,3,male,32.0,0,0,7.75,Q,Third,man,True,NaN,Queenstown,no,True


In [55]:
phool=sns.load_dataset('iris')
phool.sample(100)
# phool.head()

,sepal_length,sepal_width,petal_length,petal_width,species
40,5.0,3.5,1.3,0.3,setosa
99,5.7,2.8,4.1,1.3,versicolor
73,6.1,2.8,4.7,1.2,versicolor
100,6.3,3.3,6.0,2.5,virginica
32,5.2,4.1,1.5,0.1,setosa
...,...,...,...,...,...
138,6.0,3.0,4.8,1.8,virginica
91,6.1,3.0,4.6,1.4,versicolor
43,5.0,3.5,1.6,0.6,setosa
106,4.9,2.5,4.5,1.7,virginica


In [56]:
kashti.to_excel('kashti.xlsx')
phool.to_excel('phool.xlsx')

In [ ]:
# to save data in previous folder
phool.to_excel('../day_8/phool.xlsx')

In [61]:
#load dataset from current folder
df = pd.read_excel('kashti.xlsx')
df

,Unnamed: 0,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True
887,887,1,1,female,19.0,0,0,30.0000,S,First,woman,False,B,Southampton,yes,True
888,888,0,3,female,NaN,1,2,23.4500,S,Third,woman,False,NaN,Southampton,no,False
889,889,1,1,male,26.0,0,0,30.0000,C,First,man,True,C,Cherbourg,yes,True


In [65]:
df['embark_town']

0      Southampton
1        Cherbourg
2      Southampton
3      Southampton
4      Southampton
          ...     
886    Southampton
887    Southampton
888    Southampton
889      Cherbourg
890     Queenstown
Name: embark_town, Length: 891, dtype: object

In [66]:
df[['embark_town','survived']]

,embark_town,survived
0,Southampton,0
1,Cherbourg,1
2,Southampton,1
3,Southampton,1
4,Southampton,0
...,...,...
886,Southampton,0
887,Southampton,1
888,Southampton,0
889,Cherbourg,1
